v03_pipeline/lib/reference_datasets/dbnsfp.py

```diff
@@ -0,0 +1,83 @@
+import tempfile
+import urllib
+import zipfile
+
+import hail as hl
+
+from v03_pipeline.lib.model import ReferenceGenome
+from v03_pipeline.lib.reference_datasets.misc import key_by_locus_alleles
+
+SHARED_TYPES = {
+    'REVEL_score': hl.tfloat32,
+    'fathmm-MKL_coding_score': hl.tfloat32,
+    'MutPred_score': hl.tfloat32,
+    'PrimateAI_score': hl.tfloat32,
+}
+TYPES = {
+    ReferenceGenome.GRCh37: {
+        **SHARED_TYPES,
+        'pos(1-based)': hl.tint,
+        'CADD_phred_hg19': hl.tfloat32,
+    },
+    ReferenceGenome.GRCh38: {
+        **SHARED_TYPES,
+        'hg19_pos(1-based)': hl.tint,
+        'CADD_phred': hl.tfloat32,
+    },
+}
+
+SHARED_RENAME = {
+    'fathmm-MKL_coding_score': 'fathmm_MKL_coding_score',
+}
+RENAME = {
+    ReferenceGenome.GRCh37: {
+        **SHARED_RENAME,
+        'hg19_chr': 'chrom',
+        'hg19_pos(1-based)': 'pos',
+    },
+    ReferenceGenome.GRCh38: {
+        **SHARED_RENAME,
+        '#chr': 'chrom',
+        'pos(1-based)': 'pos',
+    },
+}
+
+PREDICTOR_SCORES = {
+    'REVEL_score',
+    'SIFT_score',
+    'Polyphen2_HVAR_score',
+    'VEST4_score',
+    'MPC_score',
+}
+PREDICTOR_FIELDS = ['MutationTaster_pred']
+
+
+def predictor_parse(field: hl.StringExpression) -> hl.StringExpression:
+    return field.split(';').find(lambda p: p != '.')
+
+
+# adapted from download_and_create_reference_datasets/v02/hail_scripts/write_dbnsfp_ht.py
+def get_ht(raw_dataset_path: str, reference_genome: ReferenceGenome) -> hl.Table:
+    types = TYPES[reference_genome]
+    rename = RENAME[reference_genome]
+
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
I think there's a subtle and annoying bug here.  This directory will get deleted when the function returns a lazy handle to the hail table.  But, when we go to actually write the table in `UpdatedReferenceDataset`, the underlying `*.gz` files will have disappeared and hail will throw an exception.  

We have a few alternatives here:
1) leave the temp files in place and pollute disk (which is what we are now doing for clinvar/mitimpact).  This is generally fine but might cause issues for large datasets that might be >100GB.
2) checkpoint to `seqr-scratch-temp` inside this function and clean up disk immediately.

I think moving towards 2) (or brainstorming something else) is probably better!

Reply:
I'm just going to share what you did for mitimpact

Reply:
👍